Mass Transfer to translateable, and translate to Finnish.

xet7 opened the pull request.

ocdtrekkie commented:
Awesome! 👍

ocdtrekkie commented:
Do we know how this falls back for other languages, @xet7? If another language doesn't have the strings yet, will they fall back to English or be unreadable?

ocdtrekkie commented:
@caasi and @urakagi may be interested in this PR too. :)

xet7 commented:
@ocdtrekkie 

I did get Sandstorm compiled #3177 , and it seems these changes do not work properly yet. I will try to get this working later.

![sandstorm-en-broken](https://user-images.githubusercontent.com/15545/72226718-cf75de00-359c-11ea-961b-210549d436d3.png)

![sandstorm-fi-broken](https://user-images.githubusercontent.com/15545/72226722-d56bbf00-359c-11ea-9d25-4803c331d8eb.png)

xet7 commented:
I fixed translations. This is how it now looks for English and French, because French has not translated Apps and Grains:

![sandstorm-en-fixed](https://user-images.githubusercontent.com/15545/72348589-6482ef00-36e3-11ea-9d71-d5d489dc861f.png)

And here is Finnish:

![sandstorm-fi-fixed](https://user-images.githubusercontent.com/15545/72348607-6fd61a80-36e3-11ea-8e09-3b216bb0095b.png)

ocdtrekkie commented:
Awesome. Though I notice the transfers-related lang strings are no longer grouped up. Is it not possible to get it working with that extra transfers group in there? (Honestly, I'm not sure why all the other language strings look so much more broken up, I'm wondering if there's more alt texts and the like in other parts of the shell that were grouped up when i18n was set up, that wasn't done with the mass transfers feature.)

xet7 commented:
@ocdtrekkie 

Yes sure it's possible. Just a moment.

The reason why in that page above that long string at bottom is broken up to multiple string is, that `<i></i>` is not allowed in JSON tranlation, I don't know why, because `<strong></strong>` is allowed.

ocdtrekkie commented:
`<i>` is supposedly deprecated, so `<strong>` may be explicitly allowed as a tag.

ocdtrekkie commented on shell/i18n/en.i18n.json:
```diff
@@ -344,44 +344,47 @@
           "hasApps": "Select an app and create your first grain.",
           "hasNoApp": "Install an app to create your first grain."
         },
-        "transfers": {
```
My question was just regarding this change. Whether or not the right fix was to remove this "grouping" for transfers, or to add it to the places where these strings are invoked.

xet7 commented:
@ocdtrekkie 

Well, at first I just removed grouping to fix it, because I was lazy.

Now I have added grouping back to these files.

Yes those JSON values can be grouped in any way. It just requires changes templates and translation JSON files.

xet7 commented:
Hmm, it seems something broke compiling Sandstorm on Debian sid, I don't know yet was it my use of Node v12 or something else. I'll try to fix it so that I could test this.
```
 link: sandstorm/util-test.o
✔ test: sandstorm/appid-replacements-test
  link: sandstorm/update-tool.o
✔ test: sandstorm/util-test
✔ test: capnp/fuzz-test
✘ link: sandstorm/run-bundle.o
    /usr/bin/ld: tmp/kj/async-io-unix.o: in function `operator()':
    /ekam-provider/canonical/kj/async-io-unix.c++:995: warning: Using '
      getaddrinfo' in statically linked applications requires at runtime the 
      shared libraries from the glibc version used for linking
✘ jstest: node-capnp/capnp-test.js
    full log: tmp/node-capnp/capnp-test.js.log
✘ install: node-capnp/node-capnp.ekam-manifest
    node-capnp/capnp.node: not found
✘ install: sandstorm/sandstorm.ekam-manifest
    sandstorm/run-bundle: not found
✘ compile: sandstorm/sandstorm-http-bridge.c++
    /ekam-provider/canonical/sandstorm/sandstorm-http-bridge.c++:1520:10: 
      warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
            [this, KJ_MVCAP(httpRequest), KJ_MVCAP(clientStream), responseStream
      , context]
             ^~~~~
    /ekam-provider/canonical/sandstorm/sandstorm-http-bridge.c++:2468:37: 
      warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
        return address->connect().then([this, loggedSlowStartupMessage](auto x) 
      -> void {
                                        ^~~~~
    In file included from /ekam-provider/canonical/sandstorm/sandstorm-http-
      bridge.c++:59:
    In file included from /ekam-provider/canonical/sandstorm/util.h:35:
    In file included from /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c+
      +/9/set:61:
    /usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/bits/stl_set.h:193
      :9: error: no matching member function for call to '_M_insert_range_unique
      '
            { _M_t._M_insert_range_unique(__first, __last); }
              ~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /ekam-provider/canonical/sandstorm/util.h:244:30: note: in instantiation of 
      function template specialization 'std::set<kj::StringPtr, std::less<kj::
      StringPtr>, std::allocator<kj::StringPtr> >::set<capnp::_::
      IndexingIterator<const capnp::List<capnp::Text, capnp::Kind::BLOB>::Reader
      , capnp::Text::Reader> >' requested here
      HeaderWhitelist(T&& list): patterns(list.begin(), list.end()) {}
                                 ^
    /ekam-provider/canonical/sandstorm/sandstorm-http-bridge.c++:154:23: note: 
      in instantiation of function template specialization 'sandstorm::
      HeaderWhitelist::HeaderWhitelist<capnp::List<capnp::Text, capnp::Kind::
    ...(log truncated; use -l to increase log limit)...
✘ link: node-capnp/capnp.o
✘ compile: sandstorm/web-session-bridge.c++
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:143:35: warning: 
      lambda capture 'method' is not used [-Wunused-lambda-capture]
          auto doNonStreaming = [this,method,path,&headers,&requestBody,&
      response]() {
                                     ~^~~~~~
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:173:24: warning: 
      lambda capture 'headers' is not used [-Wunused-lambda-capture]
              .then([this,&headers,&requestBody,&response,KJ_MVCAP(streamer)]
                         ~~^~~~~~~
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:177:11: warning: 
      lambda capture 'this' is not used [-Wunused-lambda-capture]
          }, [this,KJ_MVCAP(doNonStreaming)](kj::Exception&& e) -> kj::Promise<
      void> {
              ^~~~~
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:192:35: warning: 
      lambda capture 'method' is not used [-Wunused-lambda-capture]
          auto doNonStreaming = [this,method,path,&headers,&requestBody,&
      response]() {
                                     ~^~~~~~
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:222:24: warning: 
      lambda capture 'headers' is not used [-Wunused-lambda-capture]
              .then([this,&headers,&requestBody,&response,KJ_MVCAP(streamer)]
                         ~~^~~~~~~
    /ekam-provider/canonical/sandstorm/web-session-bridge.c++:226:11: warning: 
      lambda capture 'this' is not used [-Wunused-lambda-capture]
          }, [this,KJ_MVCAP(doNonStreaming)](kj::Exception&& e) -> kj::Promise<
      void> {
              ^~~~~
    In file included from /ekam-provider/canonical/sandstorm/web-session-bridge.
      c++:17:
    ...(log truncated; use -l to increase log limit)...
make: *** [Makefile:256: tmp/.ekam-run] Error 1
```

ocdtrekkie commented:
Thanks for re-adding the grouping! One we have more translations updated to use this, it's a lot more work to fix! Note that we discussed this PR during Office Hours and Kenton is going to try to look at it for the next release.

zenhack commented:
Quoting Lauri Ojansivu (2020-01-14 12:16:05)

>    Hmm, it seems something broke compiling Sandstorm on Debian sid, I
>    don't know yet was it my use of Node v12 or something else. I'll try to
>    fix it so that I could test this.

I think node-capnp only works on node 10 right now; there's an issue
about supporting 12 here:

    https://github.com/capnproto/node-capnp/issues/52

Also the errors from sandstorm-http-bridge look like the same ones from
before, did you remember KJ_STD_COMPAT?

xet7 commented:
@zenhack 

What does mean this node.h missing?
```
link: capnp/compat/json-test.o
✔ test: capnp/schema-loader-test
✔ test: capnp/compat/json-test
✔ test: capnp/serialize-text-test
  link: sandstorm/test-app/test-app.o
    /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace)::SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String, unsigned int, kj::_::
      NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress::LookupParams&&) const':
    async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
  link: sandstorm/sandstorm-http-bridge.o
    /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace)::SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String, unsigned int, kj::_::
      NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress::LookupParams&&) const':
    async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
  link: sandstorm/appid-replacements-test.o
✔ test: sandstorm/appid-replacements-test
  link: sandstorm/spk-standalone.o
  link: sandstorm/run-bundle.o
    /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace)::SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String, unsigned int, kj::_::
      NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress::LookupParams&&) const':
    async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
  link: sandstorm/fuse-test-bind.o
  link: sandstorm/app-index/app-index.o
  install: sandstorm/sandstorm.ekam-manifest
  link: sandstorm/update-tool.o
  link: sandstorm/util-test.o
✔ test: sandstorm/util-test
✔ test: capnp/fuzz-test
✘ compile: node-capnp/capnp.cc
    /ekam-provider/canonical/node-capnp/capnp.cc:29:10: fatal error: 'node.h' file not found
    #include <node.h>
             ^~~~~~~~
    Stack dump:
    0.	Program arguments: /home/wekan/repos/sandstorm/deps/llvm-build/Release+Asserts/bin/clang -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -mrelax-all -disable-free -disable-llvm-
      verifier -discard-value-names -main-file-name capnp.cc -mrelocation-model pic -pic-level 2 -mthread-model posix -mframe-pointer=all -fmath-errno -fno-rounding-math -masm-verbose -
      mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -dwarf-column-info -debugger-tuning=gdb -resource-dir /home/wekan/repos/sandstorm/deps/llvm-build/Release+
      Asserts/lib/clang/10.0.0 -I /ekam-provider/c++header -D KJ_STD_COMPAT -D SANDSTORM_BUILD=0 -D KJ_HAS_OPENSSL -D KJ_HAS_ZLIB -D KJ_HAS_LIBDL -I /include/node -internal-isystem /usr/lib/
      gcc/x86_64-linux-gnu/8/../../../../include/c++/8 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 -internal-isystem /usr/lib/gcc/x86_64-
      linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/backward -internal-isystem /usr/local/include -
      internal-isystem /home/wekan/repos/sandstorm/deps/llvm-build/Release+Asserts/lib/clang/10.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem 
      /include -internal-externc-isystem /usr/include -Wall -Wextra -Wglobal-constructors -Wno-sign-compare -Wno-unused-parameter -std=c++1z -fdeprecated-macro -fdebug-compilation-dir /home/
      wekan/repos/sandstorm -ferror-limit 19 -fmessage-length 0 -pthread -fgnuc-version=4.2.1 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -faddrsig -o node-
      capnp/capnp.o -x c++ /ekam-provider/canonical/node-capnp/capnp.cc 
    1.	/usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unordered_map.h:1190:72: current parser token ';'
    2.	/ekam-provider/c++header/capnp/orphan.h:28:1: parsing namespace 'capnp'
    3.	/usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unordered_map.h:33:1: parsing namespace 'capnp::std'
    #0 0x000000000247f718 (/home/wekan/repos/sandstorm/deps/llvm-build/Release+Asserts/bin/clang+0x247f718)
    clang: error: unable to execute command: Segmentation fault
    clang: error: clang frontend command failed due to signal (use -v to see invocation)
    clang version 10.0.0 (https://github.com/llvm/llvm-project/ e84b7a5fe230e42b8e6fe451369874a773bf1867)
    Target: x86_64-unknown-linux-gnu
    Thread model: posix
    InstalledDir: /home/wekan/repos/sandstorm/deps/llvm-build/Release+Asserts/bin
    clang: note: diagnostic msg: PLEASE submit a bug report to https://crbug.com and run tools/clang/scripts/process_crashreports.py (only works inside Google) which will upload a report and
      include the crash backtrace, preprocessed source, and associated run script.
    clang: note: diagnostic msg: Error generating preprocessed source(s).
✘ install: node-capnp/node-capnp.ekam-manifest
    node-capnp/capnp.node: not found
✘ jstest: node-capnp/capnp-test.js
    full log: tmp/node-capnp/capnp-test.js.log
make: *** [Makefile:256: tmp/.ekam-run] Error 1
root@sandstorm:/home/wekan/repos/sandstorm# node -v
v10.15.2
root@sandstorm:/home/wekan/repos/sandstorm# npm -v
5.8.0
root@sandstorm:/home/wekan/repos/sandstorm# cat ../build.sh 
#make clean
make CFLAGS+=-DKJ_STD_COMPAT
#sudo make update

root@sandstorm:/home/wekan/repos/sandstorm# cat /etc/issue
Debian GNU/Linux 10 \n \l
```

zenhack commented:
It's not finding the dev headers for nodejs.

I think it's supposed to actually pull these out of the meteor bundle,
rather than your system (which is why this is working on an arch box
whose system node is 13.x). What does `./find-meteor-dev-bundle.sh`
print?

(You could also try installing nodejs-dev or somesuch but I don't think
that's the right thing since meteor bundles its own version...)

Quoting Lauri Ojansivu (2020-01-14 14:43:35)
>    [1]@zenhack
>
>    What does mean this node.h missing?
> link: capnp/compat/json-test.o
> � test: capnp/schema-loader-test
> � test: capnp/compat/json-test
> � test: capnp/serialize-text-test
>   link: sandstorm/test-app/test-app.o
>     /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace):
> :SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String,
>  unsigned int, kj::_::
>       NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress:
> :LookupParams&&) const':
>     async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically
>  linked applications requires at runtime the shared libraries from the glibc ver
> sion used for linking
>   link: sandstorm/sandstorm-http-bridge.o
>     /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace):
> :SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String,
>  unsigned int, kj::_::
>       NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress:
> :LookupParams&&) const':
>     async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically
>  linked applications requires at runtime the shared libraries from the glibc ver
> sion used for linking
>   link: sandstorm/appid-replacements-test.o
> � test: sandstorm/appid-replacements-test
>   link: sandstorm/spk-standalone.o
>   link: sandstorm/run-bundle.o
>     /usr/bin/ld: tmp/kj/async-io-unix.o: in function `kj::(anonymous namespace):
> :SocketAddress::lookupHost(kj::LowLevelAsyncIoProvider&, kj::String, kj::String,
>  unsigned int, kj::_::
>       NetworkFilter&)::$_3::operator()(kj::(anonymous namespace)::SocketAddress:
> :LookupParams&&) const':
>     async-io-unix.c++:(.text+0x5383): warning: Using 'getaddrinfo' in statically
>  linked applications requires at runtime the shared libraries from the glibc ver
> sion used for linking
>   link: sandstorm/fuse-test-bind.o
>   link: sandstorm/app-index/app-index.o
>   install: sandstorm/sandstorm.ekam-manifest
>   link: sandstorm/update-tool.o
>   link: sandstorm/util-test.o
> � test: sandstorm/util-test
> � test: capnp/fuzz-test
> � compile: node-capnp/capnp.cc
>     /ekam-provider/canonical/node-capnp/capnp.cc:29:10: fatal error: 'node.h' fi
> le not found
>     #include <node.h>
>              ^~~~~~~~
>     Stack dump:
>     0.  Program arguments: /home/wekan/repos/sandstorm/deps/llvm-build/Release+A
> sserts/bin/clang -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -mrelax-all -di
> sable-free -disable-llvm-
>       verifier -discard-value-names -main-file-name capnp.cc -mrelocation-model
> pic -pic-level 2 -mthread-model posix -mframe-pointer=all -fmath-errno -fno-roun
> ding-math -masm-verbose -
>       mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -
> dwarf-column-info -debugger-tuning=gdb -resource-dir /home/wekan/repos/sandstorm
> /deps/llvm-build/Release+
>       Asserts/lib/clang/10.0.0 -I /ekam-provider/c++header -D KJ_STD_COMPAT -D S
> ANDSTORM_BUILD=0 -D KJ_HAS_OPENSSL -D KJ_HAS_ZLIB -D KJ_HAS_LIBDL -I /include/no
> de -internal-isystem /usr/lib/
>       gcc/x86_64-linux-gnu/8/../../../../include/c++/8 -internal-isystem /usr/li
> b/gcc/x86_64-linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 -internal-is
> ystem /usr/lib/gcc/x86_64-
>       linux-gnu/8/../../../../include/x86_64-linux-gnu/c++/8 -internal-isystem /
> usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/backward -internal-isys
> tem /usr/local/include -
>       internal-isystem /home/wekan/repos/sandstorm/deps/llvm-build/Release+Asser
> ts/lib/clang/10.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-
> gnu -internal-externc-isystem
>       /include -internal-externc-isystem /usr/include -Wall -Wextra -Wglobal-con
> structors -Wno-sign-compare -Wno-unused-parameter -std=c++1z -fdeprecated-macro
> -fdebug-compilation-dir /home/
>       wekan/repos/sandstorm -ferror-limit 19 -fmessage-length 0 -pthread -fgnuc-
> version=4.2.1 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-sho
> w-option -faddrsig -o node-
>       capnp/capnp.o -x c++ /ekam-provider/canonical/node-capnp/capnp.cc
>     1.  /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unordered
> _map.h:1190:72: current parser token ';'
>     2.  /ekam-provider/c++header/capnp/orphan.h:28:1: parsing namespace 'capnp'
>     3.  /usr/lib/gcc/x86_64-linux-gnu/8/../../../../include/c++/8/bits/unordered
> _map.h:33:1: parsing namespace 'capnp::std'
>     #0 0x000000000247f718 (/home/wekan/repos/sandstorm/deps/llvm-build/Release+A
> sserts/bin/clang+0x247f718)
>     clang: error: unable to execute command: Segmentation fault
>     clang: error: clang frontend command failed due to signal (use -v to see inv
> ocation)
>     clang version 10.0.0 (https://github.com/llvm/llvm-project/ e84b7a5fe230e42b
> 8e6fe451369874a773bf1867)
>     Target: x86_64-unknown-linux-gnu
>     Thread model: posix
>     InstalledDir: /home/wekan/repos/sandstorm/deps/llvm-build/Release+Asserts/bi
> n
>     clang: note: diagnostic msg: PLEASE submit a bug report to https://crbug.com
>  and run tools/clang/scripts/process_crashreports.py (only works inside Google)
> which will upload a report and
>       include the crash backtrace, preprocessed source, and associated run scrip
> t.
>     clang: note: diagnostic msg: Error generating preprocessed source(s).
> � install: node-capnp/node-capnp.ekam-manifest
>     node-capnp/capnp.node: not found
> � jstest: node-capnp/capnp-test.js
>     full log: tmp/node-capnp/capnp-test.js.log
> make: *** [Makefile:256: tmp/.ekam-run] Error 1
> root@sandstorm:/home/wekan/repos/sandstorm# node -v
> v10.15.2
> root@sandstorm:/home/wekan/repos/sandstorm# npm -v
> 5.8.0
> root@sandstorm:/home/wekan/repos/sandstorm# cat ../build.sh
> #make clean
> make CFLAGS+=-DKJ_STD_COMPAT
> #sudo make update
>
> root@sandstorm:/home/wekan/repos/sandstorm# cat /etc/issue
> Debian GNU/Linux 10 \n \l
>
>    --
>    You are receiving this because you were mentioned.
>    Reply to this email directly, [2]view it on GitHub, or [3]unsubscribe.
>
> Verweise
>
>    1. https://github.com/zenhack
>    2. https://github.com/sandstorm-io/sandstorm/pull/3175?email_source=notifications&email_token=AAGXYPRB7OPMQJMLBR5CGULQ5YIWPA5CNFSM4J6JBLP2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEI53YDQ#issuecomment-574340110
>    3. https://github.com/notifications/unsubscribe-auth/AAGXYPRZZP7XI4ECUHN2EVLQ5YIWPANCNFSM4J6JBLPQ

xet7 commented:
Well, for example on Debian sid it's this:
```
/home/user/.meteor/packages/meteor-tool/.1.8.1.1xa3774.p2bm++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/bin
```
I'm looking at my other Linux installs later.

kentonv commented on shell/client/transfers-client.js:
```diff
@@ -164,8 +164,7 @@ Template.transfers.events({
     if (!confirm(
-        "WARNING: You are giving " + destination + " permission to download all your " +
-        "grains from this server. Are you sure?")) {
+      TAPi18n.__("grains.grainlist.sandstormGrainListPage.transfers.warningGiving") + " " + destination + " " + TAPi18n.__("grains.grainlist.sandstormGrainListPage.transfers.warningPermission"))) {
```
It seems like it would make a lot of sense to use parameters here, rather than concatenate strings, so that the translation string doesn't need to be split. Similarly for some of the other translations you added. What do you think?

xet7 replied:
Well, yes, but how?

kentonv replied:
There are lots of translations that have `%s` in them... try grepping for them to see how the parameter is passed in.

I think you can just add the params to the call, like `TAPi18n.__("translationName", param1, param2)`.

xet7 replied:
I changed this to use parameters.

xet7 commented:
> `<i>` is supposedly deprecated, so `<strong>` may be explicitly allowed as a tag.

`<strong>` was not allowed anymore, so I removed that emphasis.

xet7 commented:
@kentonv @ocdtrekkie 

I have tested this works now.

xet7 commented:
![sa-fi](https://user-images.githubusercontent.com/15545/74110316-2074f400-4b94-11ea-8b14-2a6f4b01a8d5.png)

![sa-fi2](https://user-images.githubusercontent.com/15545/74110318-27036b80-4b94-11ea-9237-ba683280e316.png)

xet7 commented:
![sa-fi4](https://user-images.githubusercontent.com/15545/74110417-1ef7fb80-4b95-11ea-8a7f-8ab6570b83a3.png)

![sa-fi3](https://user-images.githubusercontent.com/15545/74110420-24554600-4b95-11ea-8dda-e1a3f29c5a98.png)

ocdtrekkie commented on shell/i18n/en.i18n.json:
```diff
@@ -344,7 +345,42 @@
+          "whatDoesThisDo": "What does this do?",
+          "whatCan": "You can use this function to transfer all of your grains to a new Sandstorm server. This automates the process of saving a backup of each grain on this server, then restoring each backup on the server you specify.",
+          "whatWhen": "When you initiate the transfer above, you will be redirected to the destination server where you will be prompted to sign in. You will then be given the option to choose which grains to transfer, and then finally you will initiate the transfer. The transfer will continue in the background; you can log off.",
+          "whatOrder": "In order for a transfer to work, the destination server must be able to reach the source server directly over the internet. However, the reverse is not necessary. Thus, you can perform a mass trasfer that transfers grains from a server on the public internet (like Sandstorm Oasis) to a server on a private internal network, but not vice versa. If the source server supports  ...
```
s/trasfer/transfer

This spelling error occurs in the original text, and I just found it here. But let's fix it while we're at it.

ocdtrekkie commented on shell/i18n/en.i18n.json:
```diff
@@ -344,7 +345,42 @@
+          "inOrder": "In order to start a new transfer (incoming or outgoing), you must cancel the existing one.",
+          "cancelTransfer": "Cancel Transfer",
+          "initiateInboundTransfer": "Initiate Inbound Tranfer",
+          "initiateOutboundTransfer": "Initiate Outbound Tranfer",
```
Here on both `initiateInboundTransfer` and `initiateOutboundTransfer`, you introduced a new misspelling of transfer, Tranfer.

ocdtrekkie commented:
I am a bit confused because there are plenty of other <strong> tags in the language files, and they appear, at least, to work. What happens if you leave them in? Do things just not bold correctly, or are they appearing as `<strong>` in the UI?

My impression is if they don't appear at all, leave them in and we should consider it a bug that they don't work. I am iffy on removing them as Kenton presumably used them there to be helpful to people trying to use the feature.

xet7 commented:
@ocdtrekkie 

It appears as text `<strong>` in UI. It does not make text bold.

ocdtrekkie commented:
So, I have a theory, knowing nothing about our i18n library:

https://github.com/sandstorm-io/sandstorm/blob/6607cc8ebb35bc405ce4ccbe877363d826df92c5/shell/client/shell.html#L411-L413

L413 has an extra curly brace around it, and has a Strong tag. And sure enough, in the readme at https://github.com/TAPevents/tap-i18n if you search for `{{{`, you'll see "Note that we use triple {{{ for html rendering".

xet7 commented:
![curly](https://user-images.githubusercontent.com/15545/74169230-aa27ce80-4c33-11ea-82bd-9197f45bf18d.png)

ocdtrekkie commented:
Looks good to me.

xet7 commented:
![curly2](https://user-images.githubusercontent.com/15545/74171657-941c0d00-4c37-11ea-9834-50688bc50e64.png)

![curly1](https://user-images.githubusercontent.com/15545/74171688-9d0cde80-4c37-11ea-847a-4e4bc549b67a.png)

kentonv commented:
Thanks!
